bundles/org.eclipse.equinox.preferences/OSGI-INF/org.eclipse.core.internal.preferences.BundleStateScopeServiceFactory.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I'm confused by the use of annotations and having this `component.xml` in the source code.  Shouldn't this file be generated by the annotations?

Reply:
The file is generated by the PDE DS plugin in the IDE, but generation in the tycho build will only be possible since the new Tycho 2.7.2 / 3.x release.

Reply:
If you modify the `.java` source with the annotations then PDE will write the file to this source code location?  Perhaps that is alright.  Does PDE complain if you try to hand edit this XML and it conflicts with what the source annotations specify?

Reply:
Aren't all platform repositories already build with the Tycho 3.0.0-SNAPSHOT?
So the component's xml files could already be ignored in git, couldn' t they?

Reply:
There seem to be some back and forth switching, so I really don't like to rely on unreleased features here, also the API tools don't work without the import just crashing with code 13 (no idea what this is about), there are several component.xmls already in the platform repos so no need to be extra concerned here... as mentioned above if DS seems not suitable here I can rewrite this with a servcie-tracker and a regular service-factory ...

Reply:
> There seem to be some back and forth switching, so I really don't like to rely on unreleased features here, also the API tools don't work without the import just crashing with code 13 (no idea what this is about), there are several component.xmls already in the platform repos so no need to be extra concerned here...

Sure no objection against that. I just wanted to suggest/mention it. :)